### PR TITLE
docs: Propose initial v0.10 migration guide with JSON updates

### PR DESCRIPTION
**Pull request by tim775**

### CODE REVIEW

File: docs/guides/v0.10_migration.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+🚀 We've updated our JSON output format in Infracost `v0.10.0`.
+
+## Migration guide
+
+### JSON output
+
+The JSON output from `infracost breakdown --path=path/to/code --format json` has been changed to the following format:
+
+* `version` has been bumped from `v0.2` to `v0.3`.
```
Comment (aliscott) [inline docs/guides/v0.10_migration.md:20]:
```suggestion
* `version` has been bumped from `0.2` to `0.3`.
```

File: docs/guides/v0.10_migration.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+* `resources` now have `name`, `type`, and `address` fields.  Previously the `name` field contained the address.  This change follows the convetion in the terraform plan json.
```
Comment (aliscott) [inline docs/guides/v0.10_migration.md:21]:
```suggestion
* `resources` now have `name`, `type`, and `address` fields.  Previously the `name` field contained the address.  This change follows the convention in the Terraform plan json.
```

File: docs/guides/v0.10_migration.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      },
+      "diff": {
+        "resources": [...],
+        "totalMonthlyCost": 730,
+        "totalHourlyCost": 1
+      }
+    }
+  ],
+  "totalMonthlyCost": 2920,
+  "totalHourlyCost": 4,
```
Comment (aliscott) [inline docs/guides/v0.10_migration.md:74]: We've also added `pastTotalXCost` and `diffTotalXCost` here so should probably add them to this example as well.

File: docs/guides/v0.10_migration.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+          {
+            "address": "aws_instance.web_app",
+            "type": "aws_instance",
+            "name": "web_app",
+            "metadata": {},
+            "hourlyCost": 1.017315068493150679,
+            "monthlyCost": 742.64,
+            "costComponents": [
+              {
+                "name": "Instance usage (Linux/UNIX, on-demand, m5.4xlarge)",
```
Comment (aliscott) [inline docs/guides/v0.10_migration.md:47]: Are we wanting to take this opportunity to also separate out the brackets into labels or some other structured format?

Comment (alikhajeh1) [reply docs/guides/v0.10_migration.md:47]: +1, key=value pairs might be most explicit, e.g. instance_type=m5.4xlarge, purchase_option=on-demand. We could still render it in the UI by comma-separating the values but having the keys enables other display formats in the future (and in the UI).

File: docs/guides/v0.10_migration.md (hunk 2 above)
Comment (alikhajeh1) [inline docs/guides/v0.10_migration.md:21]: Do `name`, `type`, and `address` also work for Pulumi, CloudFormation and Azure ARM?
Can we please also add something like this to show the before/after? Did I get the module example correct?

The following two examples illustrate the change:
1. The old `name: aws_instance.web_app` will now be stored as `type: aws_instance, name: web_app, address: aws_instance.web_app`.
2. For modules, the old `name: module.my_api.module.metric_alarm_target_4xx.aws_cloudwatch_metric_alarm.this[0]` will now be stored as `type: aws_cloudwatch_metric_alarm, name: this[0], address: module.my_api.module.metric_alarm_target_4xx.aws_cloudwatch_metric_alarm.this[0]`.

File: docs/guides/v0.10_migration.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+The JSON output from `infracost breakdown --path=path/to/code --format json` has been changed to the following format:
+
+* `version` has been bumped from `v0.2` to `v0.3`.
+* `resources` now have `name`, `type`, and `address` fields.  Previously the `name` field contained the address.  This change follows the convetion in the terraform plan json.
+* Fields containing prices, costs, and quantities are now numbers instead of strings.  
+* A [JSON schema](http://json-schema.org/) for the JSON output is now published at https://github.com/infracost/infracost/blob/master/schema/infracost.schema.json.
+
+```json
+{
+  "version": "0.3",
```
Comment (alikhajeh1) [inline docs/guides/v0.10_migration.md:27]: no currency?

File: docs/guides/v0.10_migration.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+* `version` has been bumped from `0.2` to `0.3`.
+* `resources` now have `name`, `type`, and `address` fields.  Previously the `name` field contained the address.  This change follows the convention in the Terraform plan json.  For example:
+  * `{ ... "name": "aws_instance.web_app" ...}` will now be stored as: `{ ... "type": "aws_instance", "name": "web_app", "address": "aws_instance.web_app" ... }`
+  * `{ ... "name": "module.my_api.module.metric_alarm_target_4xx.aws_cloudwatch_metric_alarm.this[0]" ...}` will now be stored as `{ ... "type": "aws_cloudwatch_metric_alarm", "name": "this[0]", "address": "module.my_api.module.metric_alarm_target_4xx.aws_cloudwatch_metric_alarm.this[0]" ... }`
+* Fields containing prices, costs, and quantities are now numbers instead of strings.  
+* When possible, cost component configuration details will be expanded as a key value list.  `{
+  "name": "Instance usage (Linux/UNIX, on-demand, m5.4xlarge)" ...}` will now be stored as `{
+  "name": "Instance usage (Linux/UNIX, on-demand, m5.4xlarge)", "parameters": { "platform": "Linux/UNIX", "purchase_option": "on-demand", "instance_type": "m5.4xlarge" } ... }`
```
Comment (aliscott) [inline docs/guides/v0.10_migration.md:27]: If we want a parameters field, should we remove the brackets from the name?

**Comment by alikhajeh1** [comment]
Thanks @tim775! I've created https://github.com/infracost/infracost/issues/1260 so users can provide feedback on this

**Pull request closed without merging**